graph/src/components/link_resolver/ipfs.rs

```diff
@@ -43,70 +45,51 @@ fn retry_policy<I: Send + Sync>(
 /// of clients where hopefully one already has the file, and just get the file
 /// from that.
 ///
-/// The strategy here then is to use a stat API as a proxy for "do you have the
+/// The strategy here then is to cat a single byte as a proxy for "do you have the
 /// file". Whichever client has or gets the file first wins. This API is a good
 /// choice, because it doesn't involve us actually starting to download the file
 /// from each client, which would be wasteful of bandwidth and memory in the
-/// case multiple clients respond in a timely manner. In addition, we may make
-/// good use of the stat returned.
-async fn select_fastest_client_with_stat(
+/// case multiple clients respond in a timely manner.
+async fn select_fastest_client(
     clients: Arc<Vec<IpfsClient>>,
     logger: Logger,
-    api: StatApi,
     path: String,
     timeout: Duration,
     do_retry: bool,
-) -> Result<(u64, IpfsClient), Error> {
+) -> Result<IpfsClient, Error> {
     let mut err: Option<Error> = None;
 
-    let mut stats: FuturesUnordered<_> = clients
+    let mut exists: FuturesUnordered<_> = clients
         .iter()
         .enumerate()
         .map(|(i, c)| {
             let c = c.cheap_clone();
             let path = path.clone();
-            retry_policy(do_retry, "IPFS stat", &logger).run(move || {
+            retry_policy(do_retry, "IPFS exists", &logger).run(move || {
                 let path = path.clone();
                 let c = c.cheap_clone();
-                async move {
-                    c.stat_size(api, path, timeout)
-                        .map_ok(move |s| (s, i))
-                        .await
-                }
+                async move { c.exists(&path, Some(timeout)).map_ok(|()| i).await }
             })
         })
         .collect();
 
-    while let Some(result) = stats.next().await {
+    while let Some(result) = exists.next().await {
         match result {
-            Ok((stat, index)) => {
-                return Ok((stat, clients[index].cheap_clone()));
+            Ok(index) => {
+                return Ok(clients[index].cheap_clone());
             }
             Err(e) => err = Some(e.into()),
         }
     }
 
     Err(err.unwrap_or_else(|| {
         anyhow!(
-            "No IPFS clients were supplied to handle the call to object.stat. File: {}",
+            "No IPFS clients were supplied to handle the call. File: {}",
```

Comment:
That error message might be a little clearer if it said 'Could not find an IPFS client that has file {path}'